### ISSUE TITLE
Multiple Database error

### ISSUE THREAD

**Issue by dirkk0** [msiemens/tinydb#72]
Hi,

  this code ( https://gist.github.com/dirkk0/407bcf506568e1cdf20b ) implements two sequencers, seq1 and seq2. seq1 looks every second at a database stack.json, and once an entry appears, it inserts the entry with a timestamp into a dabase process.json and removes the entry from the stack. seq2 looks at process.json and processes the entries once their time is come.

seqbad does exactly the same thing as seq1 but opens both databases once in the beginning (which should be the right way to do it). Now, if you add entries with add.py multiple times, the old entries reappear in process.json the more you add.

Best,
  Dirk

**Comment by msiemens** [comment]
Am I correct in assuming that you're running the Python scripts in parallel from multiple Python instances?

**Comment by dirkk0** [comment]
Yes?

**Comment by msiemens** [comment]
Okay, that's propably the problem. TinyDB isn't designed to be used from multiple Python instances in parallel. We don't even support usage from multiple threads in one Python instance. I don't think we can solve this issue without adding a lot of complexity by adding some sort of read/write locking.

**Comment by eugene-eeo** [comment]
@dirkk0 I would recommend writing a storage that uses file locking. If you can't access [``fcntl``](https://en.wikipedia.org/wiki/File_locking)-like syscalls then maybe you can try a more naive approach: E.g. creating a lockfile, then deleting it when a modification is done and having your other processes waiting until your lockfile is deleted then you can achieve parallelism.

Although I hope you'd opt for a more complex route and perhaps direct all read/write requests to a single DB process using some form of IPC (e.g. [XMLRPC](https://docs.python.org/2/library/xmlrpclib.html)).

**Comment by dirkk0** [comment]
Thanks for the reply! Since re-opening the database works for me, and efficiency is not an issue in this particular case, everything is fine for me.

I still consider this a bug, though. So feel free to close this issue, if you think different.

Thanks again,
  Dirk

**Comment by msiemens** [comment]
I think this is a *wontfix* kind of bug as it is due to TinyDB's design. Nevertheless this should be documented better.

**Comment by msiemens** [comment]
Commit 58eb3ddd949599eafebdc7e5b21e7976c6dc7af7 adds a small note that using TinyDB from multiple processes is not supported.
